samcli/commands/init/init_templates.py

```diff
@@ -112,14 +105,23 @@ def _check_app_template(entry: Dict, app_template: str) -> bool:
         return bool(entry["appTemplate"] == app_template)
 
     def init_options(self, package_type, runtime, base_image, dependency_manager):
-        if not self.clone_attempted:
-            self._clone_repo()
-        if self.repo_path is None:
+        if not self._git_repo.clone_attempted:
+            shared_dir: Path = global_cfg.config_dir
```

Comment:
you use `clone_dir` in `git_repo`, maybe we should use `clone_dir` to make it consistent?

Reply:
it has a different meaning in each context:
clone-method (abstract meaning): the location to clone into
here: we are cloning SAM's templates into SAM's shared directory, a directory that is shared between all of the generated SAM applications.